File: single-number.js
/**
 * Given a non-empty array of integers, nums, where every element appears
 * twice except for one, returns that single element.
 * 
 * @param {number[]} nums - A non-empty array of integers
 * @returns {number} - Returns the element that appears only once
 */
function singleNumber (nums) {
  let numCount = {};
  for (let i = 0; i < nums.length; i++) {
    numCount[nums[i]] = (numCount[nums[i]] || 0) + 1;
  }
  for (const key in numCount) {
    if (numCount[key] === 1) {
      return Number(key);
    }
  }
}

File: src/algorithms/single-number.py
def single_number(nums):
    """Given a non-empty array of integers nums, every element appears twice 
    except for one. Find that single one. You must implement a solution with a 
    linear runtime complexity and use only constant extra space.

    Args:
        nums (list[int]): A list of integers.

    Returns:
        int: Returns the element that appears only once in nums.
    """
    seen = {}
    for num in nums:
        if num in seen:
            seen[num] += 1
        else:
            seen[num] = 1
    for key, val in seen.items():
        if val == 1:
            return key
